Userland/Libraries/LibWeb/Bindings/MainThreadVM.cpp

```diff
@@ -308,8 +309,8 @@ JS::VM& main_thread_vm()
         auto& custom_data = *verify_cast<WebEngineCustomData>(vm->custom_data());
         custom_data.root_execution_context = MUST(JS::Realm::initialize_host_defined_realm(
             *vm, [&](JS::Realm& realm) -> JS::Object* {
-                custom_data.internal_window_object = JS::make_handle(*HTML::Window::create(realm));
-                return custom_data.internal_window_object.cell();
+                auto global = JS::GlobalObject::create(realm, nullptr);
```

Comment:
Wait, this is not how you make a global object - is this just calling `Object::create()` indirectly, and does that work as expected?

Reply:
I'm not sure if we even need to have a global object in this realm for what it's used for. (Though I think that would raise even more eyebrows!) All that we need this realm and execution context for is 

1) a few places in Fetch IDL land and

 2) CSS parsing

Where we need to magically get a way to allocate JS objects, and we need to be able to attach prototypes for bound objects to *some* intrinsics object (CSS case).

Everywhere else that reaches for this top level execution context has(?) been moved to getting a realm from the caller, a parent object, an element, a document, or a browsing context.

I can change it to properly create a global object, but I think the only place that really checks the nature of this object in my testing that could hit this is the `queue_global_task` on the user interaction task source for visibility (which still looks sus).

 There seems to be a few places that say "if the global object is a window do X, else skip these steps"... But they might be thinking that the only other option is a WorkerGlobalScope??

Reply:
After all that rambling, I'm going to change this to `initialize_host_defined_realm(vm, nullptr, nullptr)` to let LibJS handle it via step 7 of https://tc39.es/ecma262/#sec-initializehostdefinedrealm